consul/consul.go

```diff
@@ -30,10 +33,29 @@ func New(config ConsulConfig) *Consul {
 }
 
 func (c *Consul) GetServices(name string) ([]*consulapi.CatalogService, error) {
-	agent, err := c.agents.GetAnyAgent()
-	if err != nil {
-		return nil, err
+	return c.getServicesUsingProviderWithRetriesOnAgentFailure(func(agent *consulapi.Client) ([]*consulapi.CatalogService, error) {
+		return c.getServicesUsingAgent(name, agent)
+	})
+}
+
+func (c *Consul) getServicesUsingProviderWithRetriesOnAgentFailure(provide ServicesProvider) ([]*consulapi.CatalogService, error) {
+	var services []*consulapi.CatalogService = nil
+	for {
```

Comment:
why? :) 
They're limited to the size of agents cache. Usually a single failing agent should be enough.